### PR TITLE
JIT: Efficient vector

### PR DESCRIPTION
**Pull request by chriseth**
Closes https://github.com/powdr-labs/powdr/issues/1854

### CODE REVIEW

**Comment by chriseth** [comment]
Compare the benchmark with https://github.com/powdr-labs/powdr/pull/1855 which does not use Arc.

**Comment by chriseth** [comment]
Some statistics:

After this PR:
```
evaluator-benchmark/sort_2700 time:   [124.46 ms 124.63 ms 124.81 ms]
jit-benchmark/sort_2700 time:   [18.175 ms 18.189 ms 18.205 ms]
```

**Review by lvella — APPROVED** [review]

File: jit-compiler/src/compiler.rs [hunk 1]
```diff
@@ -111,6 +134,18 @@ impl Add for ibig::IBig {
     fn add(a: Self, b: Self) -> Self { a + b }
 }
 
+impl<T: Clone> Add for PilVec<T> {
+    fn add(a: Self, b: Self) -> Self {
+        // TODO for a regular "push" or array::map this is very slow.
+        // We could optimize this by sharing a larger backing vector
+        // across prefix instances, allowing to extend the backing vector if
+        // our view is the full vector.
```
Comment (lvella) [inline jit-compiler/src/compiler.rs:142]: Maybe it should implement "push" and "extend", but then it would have to be Arc<RwLock<Vec<T>>>.

Comment (chriseth) [reply jit-compiler/src/compiler.rs:142]: Yeah, we can do that later, after we checked what the performance bottlenecks actually are.

**Pull request merged**